src/executables/csql.c

```diff
@@ -3330,3 +3358,161 @@ get_host_ip (unsigned char *ip_addr)
 
   return 0;
 }
+
+static int
+csql_connect (char *argument, CSQL_ARGUMENT * csql_arg)
+{
+  /*dbname can be stored DB name+ @ + remote hostname + \0 */
+  char dbname[MAXLOGNAME + CUB_MAXHOSTNAMELEN + 2];
+  char username[DB_MAX_USER_LENGTH + 1];
+  const char *delim = " @\n";
+  char *save_ptr_strtok = NULL;
+  char *db_name_ptr = NULL;
+  char *user_name_ptr = NULL;
+  char *host_name_ptr = NULL;
+  char *p = NULL;
+  int error_code;
+  CSQL_ARGUMENT csql_new_arg;
+
+  /*verify @hostname */
+  host_name_ptr = strchr (argument, '@');
+  if (host_name_ptr != NULL)
+    {
+      if (*(host_name_ptr + 1) == '\0')
+	{
+	  host_name_ptr = NULL;
+	  csql_Error_code = CSQL_ERR_CONNECT;
+	  return DO_CMD_FAILURE;
+	}
+    }
```

Comment:
I think two ifs can be merged into one, and **host_name_ptr = NULL;** is not required.: 
 if (host_name_ptr != NULL && *(host_name_ptr + 1) == '\0')
  {
    csql_Error_code = CSQL_ERR_CONNECT;
    return DO_CMD_FAILURE;
   }

Reply:
Thank you. It will be better.